awslabs/aws-lambda-powertools-typescript
# fix(logger): change logging to use stdout

## Description of your changes

As reported by user @olanb7 on the community Slack and in #747 before this PR Logger would output logs like this: 
```
2022-04-08T10:58:44.811Z	b6511ee9-4873-404e-b60c-a23cb45d3ff2	INFO {"cold_start":false,"function_arn":"arn:aws:lambda:eu-west-2:12345:function:test","function_memory_size":1024,"function_name":"test","function_request_id":"b6511ee9-4873-404e-b60c-a23cb45d3ff2","level":"INFO","message":"test message","service":"test","timestamp":"2022-04-08T10:58:44.811Z"}
```

These logs, although containing a structured JSON, are not JSON as they are always prefixed by `2022-04-08T10:58:44.811Z	b6511ee9-4873-404e-b60c-a23cb45d3ff2	INFO`. 

This also confirmed [by a line in the Lambda docs](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-logging.html#node-logging-output) (emphasis mine):
> The Node.js runtime logs the START, END, and REPORT lines for each invocation. **It adds a timestamp, request ID, and log level to each entry logged by the function**.

This behaviour works well with CloudWatch but the presence of these prefixes was an issue for customers wanting to use these structured logs elsewhere (both 3rd party services and AWS Managed Grafana).

From this PR onwards Logger will be using its own internal and private version of the [`Console` class](https://nodejs.org/api/console.html) that only outputs the structured JSON log without any prefix:
```
{"cold_start":false,"function_arn":"arn:aws:lambda:eu-west-2:12345:function:test","function_memory_size":1024,"function_name":"test","function_request_id":"b6511ee9-4873-404e-b60c-a23cb45d3ff2","level":"INFO","message":"test message","service":"test","timestamp":"2022-04-08T10:58:44.811Z"}
```

Aside from 3-lines change in the Logger class, all the rest of the changes in this PR are related to the unit and integration tests.

### How to verify this change

See the results of checks inside this PR and e2e tests (to be run).

Also see screenshot below:
![image](https://user-images.githubusercontent.com/7353869/163231163-841852c0-413e-44e6-bbdd-321ff4bbd424.png)

### Related issues, RFCs

[#747](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/747)

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist


### Breaking change checklist

N/A

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
